lib/pure/asyncdispatch.nim

```diff
@@ -1091,10 +1108,17 @@ else:
       setGlobalDispatcher(newDispatcher())
     result = gDisp
 
-  proc register*(fd: AsyncFD) =
+  proc getIoHandler*(disp: PDispatcher): Selector[AsyncData] =
+    return disp.selector
+
+  proc register*(fd: cint | SocketHandle | AsyncFD): AsyncFD {.discardable.} =
```

Comment:
Questionable API design: It either returns AsyncFD(0) or ``fd`` which I already know. Return a discardable bool instead?

Reply:
I decided to give the type ``AsyncFD`` a meaning: it is an FD that is registered in the async dispatcher. This is why this function returns an ``AsyncFD``.

Reply:
Why can I register an AsyncFD then? Am I allowed to register things multiple times?

Reply:
For backward compatibility. [If you pass in an AsyncFD then you won't get an error if its already in the dispatcher](https://github.com/nim-lang/Nim/pull/6796/files#diff-e7d29941cb2ca6bf4e0f5c627239bc7bR277)